sideboard/config.py

```diff
@@ -9,14 +10,72 @@
 from validate import Validator
 
 
+_defaults_re = re.compile(r'(.+)-defaults(\.\w+)$')
+
+
+def uniquify(x):
+    """
+    Returns an ordered copy of `x` with duplicate items removed.
+
+    >>> uniquify(['a', 'b', 'a', 'c', 'a', 'd', 'a', 'e'])
+    ['a', 'b', 'c', 'd', 'e']
+
+    """
+    seen = set()
+    return [i for i in x if i not in seen and not seen.add(i)]
+
+
 class ConfigurationError(RuntimeError):
     pass
 
 
+def get_config_overrides():
+    """
+    Returns a list of config file paths used to override the default config.
+
+    The SIDEBOARD_CONFIG_OVERRIDES environment variable may be set to a
+    semicolon separated list of absolute and/or relative paths. If the
+    SIDEBOARD_CONFIG_OVERRIDES is set, this function returns a list of its
+    contents, split on semicolons::
+
+        # SIDEBOARD_CONFIG_OVERRIDES='/absolute/config.ini;relative/config.ini'
+        return ['/absolute/config.ini', 'relative/config.ini']
+
+
+    If any of the paths listed in SIDEBOARD_CONFIG_OVERRIDES ends with the
+    suffix "<FILENAME>-defaults.<EXT>" then a similarly named path
+    "<FILENAME>.<EXT>" will also be included::
+
+        # SIDEBOARD_CONFIG_OVERRIDES='test-defaults.ini'
+        return ['test-defaults.ini', 'test.ini']
+
+
+    If the SIDEBOARD_CONFIG_OVERRIDES environment variable is NOT set, this
+    function returns a list with two relative paths::
+
+        return ['development-defaults.ini', 'development.ini']
+
+    """
+    config_overrides = os.environ.get(
+        'SIDEBOARD_CONFIG_OVERRIDES',
+        'development-defaults.ini')
+
+    config_paths = []
+    for config_path in uniquify([s.strip() for s in config_overrides.split(';')]):
+        config_paths.append(config_path)
+        m = _defaults_re.match(config_path)
```

Comment:
This (specifically _defaults_re) is some crazy regexp but I tested it in the IDE and it seems to work.